meson.build

```diff
@@ -0,0 +1,1207 @@
+project('pjproject', ['c', 'cpp', 'objc'], version: '2.13.1')
+
+# [+] - config.status: creating pjlib/include/pj/compat/os_auto.h
+# [+] - config.status: creating pjlib/include/pj/compat/m_auto.h
+# [+] - config.status: creating pjmedia/include/pjmedia/config_auto.h
+# [+] - config.status: creating pjmedia/include/pjmedia-codec/config_auto.h
+# [+] - config.status: creating pjsip/include/pjsip/sip_autoconf.h
+
+# Use compiler
+compiler = meson.get_compiler('c')
+
+# Detect compile OS system: 'darwin', ...
+os = build_machine.system()
+
+# pjlib - os_auto.h
+os_auto = configuration_data()
+os_auto.set_quoted('PJ_OS_NAME', os)
+if os == 'darwin'
+    os_auto.set('PJ_DARWINOS', 1)
+elif os == 'linux'
+    os_auto.set('PJ_LINUX', 1)
+endif
+if compiler.has_header('arpa/inet.h')
+    os_auto.set('PJ_HAS_ARPA_INET_H', 1)
+endif
+if compiler.has_header('assert.h')
+  os_auto.set('PJ_HAS_ASSERT_H', 1)
+endif
+os_auto.set('PJ_HAS_CTYPE_H', 1)
+if compiler.has_header('errno.h')
+    os_auto.set('PJ_HAS_ERRNO_H', 1)
+endif
+os_auto.set('PJ_HAS_FCNTL_H', 1)
+if compiler.has_header('limits.h')
+  os_auto.set('PJ_HAS_LIMITS_H', 1)
+endif
+#undef PJ_HAS_LINUX_SOCKET_H */
+if compiler.has_header('malloc.h')
+  os_auto.set('PJ_HAS_MALLOC_H', 1)
+endif
```

Comment:
meson's own tools/ac_converter.py will do this:
```meson
check_headers = [
    'arpa/inet.h',
    'assert.h',
    'errno.h',
    'limits.h',
    'malloc.h',
]

foreach h : check_headers
    if cc.has_header(h)
        cdata.set('HAVE_' + h.underscorify().to_upper(), 1)
    endif
endforeach
```

Minor style tweaks aside, this can be very helpful in the general case by reducing the complexity of adding an additional header check.

Reply:
I completely agree. Let's use this approach. The original PR was only in the dumb output to meson transaltion. Of course you offer great improvements.